mava/wrappers/jumanji.py

```diff
@@ -62,3 +55,87 @@ def observation_spec(self) -> specs.Spec[Observation]:
             "step_count",
         )
         return self._env.observation_spec().replace(step_count=step_count)
+
+
+class RWAREWrapper(MultiAgentWrapper):
+    """Multi-agent wrapper for the Robotic Warehouse environment."""
+
+    def __init__(self, env: RobotWarehouse):
+        super().__init__(env)
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for the Robotic Warehouse environment."""
+        n_agents = self._env.num_agents
+        observation = Observation(
+            agents_view=timestep.observation.agents_view,
+            action_mask=timestep.observation.action_mask,
+            step_count=jnp.repeat(timestep.observation.step_count, n_agents),
+        )
+        reward = jnp.repeat(timestep.reward, n_agents)
+        discount = jnp.repeat(timestep.discount, n_agents)
+        return timestep.replace(observation=observation, reward=reward, discount=discount)
+
+
+class LBFWrapper(MultiAgentWrapper):
+    """
+     Multi-agent wrapper for the Level-Based Foraging environment.
+
+    Args:
+        env (Environment): The base environment.
+        aggregate_rewards (bool): If True, aggregates rewards across agents.
+        reward_aggregation_function (str): The function for aggregating rewards ("sum" or "mean").
+    """
+
+    def __init__(
+        self,
+        env: LevelBasedForaging,
+        aggregate_rewards: bool = False,
+        reward_aggregation_function: str = "sum",
+    ):
+        super().__init__(env)
+        self._env: LevelBasedForaging
+        self._aggregate_rewards = aggregate_rewards
+        self.reward_aggregation_function = reward_aggregation_function
+        self._num_agents = self._env.num_agents
+        # Mapping aggregation functions
+        self.aggregation_functions: Dict[str, Callable] = {
+            "sum": jnp.sum,
+            "mean": jnp.mean,
+            # Add more functions as needed
+        }
+        self.aggregate_function: Callable = self.aggregation_functions.get(
+            reward_aggregation_function, jnp.sum
+        )
```

Comment:
Bit cleaner

```suggestion

        reward_aggregation_fn: Optional[Callable[[Array], Array] = jnp.sum,
    ):
        super().__init__(env)
        self._env: LevelBasedForaging
        self._num_agents = self._env.num_agents
        self._aggregate_rewards = reward_aggregation_fn is not None
        self.aggregate_function = reward_aggregation_fn
```

Reply:
I set the `jnp.sum` as the only function we'll use as @RuanJohn mentioned, I hope the current changes are better now 🙏

Reply:
even better!